cmd/commands/function.go

```diff
@@ -65,16 +66,31 @@ func FunctionCreate(buildpackBuilder core.Builder, fcTool *core.Client) *cobra.C
 				return err
 			}
 
-			if createFunctionOptions.Image == "" {
+			if createFunctionOptions.Image == "" || strings.HasPrefix(createFunctionOptions.Image, "_") {
 				prefix, err := (*fcTool).DefaultBuildImagePrefix(createFunctionOptions.Namespace)
 				if err != nil {
 					return fmt.Errorf("unable to default image: %s", err)
 				}
 				if prefix == "" {
-					return fmt.Errorf("required flag(s) \"image\" not set, this flag is optional if --image-prefix is specified during namespace init")
+					if createFunctionOptions.Image == "" {
+						return fmt.Errorf("required flag(s) \"image\" not set, this flag is optional if --image-prefix is specified during namespace init")
+					}
+					return fmt.Errorf("--image flag must include a repository, the image prefix was not set during namespace init")
+				}
+
+				fnName := args[functionCreateFunctionNameIndex]
+				if createFunctionOptions.Image == "" {
+					// combine prefix and function name to provide default image
+					createFunctionOptions.Image = fmt.Sprintf("%s/%s", prefix, fnName)
+				} else if strings.HasPrefix(createFunctionOptions.Image, "_:") {
+					// add the prefix to the specified image tag
+					createFunctionOptions.Image = strings.Replace(createFunctionOptions.Image, "_", fmt.Sprintf("%s/%s", prefix, fnName), 1)
+				} else if strings.HasPrefix(createFunctionOptions.Image, "_/") {
```

Comment:
It makes sense but I find the overloading of `_` a bit confusing at first.
It would be nice to be able to use explicit variables instead.

Here is a short example of what one could do (based on Golang [`text/template`](https://golang.org/pkg/text/template/) package):
```
imageTemplate, err := template.New("image").Parse(createFunctionOptions.Image)
if err != nil {
	return err
}
builder := &strings.Builder{}
err = imageTemplate.Execute(builder, map[string]string{
	"ImagePrefix": prefix,
	"FunctionName":    fnName,
})
if err != nil {
	return err
}
createFunctionOptions.Image = builder.String()
```

The default value for the image option could become: `{{.ImagePrefix}}/{{.FunctionName}}` and users could reference either of the variables (and arguably shoot themselves in the foot).

Reply:
I think a template string is generally too complex for users at a terminal and doesn't add enough value for scripts that can do their own substitution. If others have strong feelings, I'm willing to be convinced.

My first though was to do something more like `_prefix/foo` where `_prefix` is substituted, but then I released that 'prefix' isn't adding any value, so I removed it.

Reply:
Even without going to a full-fledged template string, I'd tend to use two different symbols instead of just one. One symbol would be for the prefix, one for the prefix+function. WDYT?

Reply:
I think adding two different symbols make the replacement more arcane and increases cognitive load as you have to think about which symbol to use. Using a single symbol to indicate replacement and using the secondary symbol (`/` or `:`) as part of the image in its natural form.

As is, the leading `_` is replaced with a value that will make the rest of the image valid. If we think the behavior is confusing, I'm open to dropping the `_:` form, re-typing the function name shouldn't add very much of a cognitive load, it was also a late addition once I saw how easy it would be to implement.

Reply:
👍